crates/fuel-core/src/service/adapters/sync.rs

```diff
@@ -83,3 +86,34 @@ impl ConsensusPort for VerifierAdapter {
         Ok(self.block_verifier.verify_consensus(header))
     }
 }
+
+#[async_trait::async_trait]
+impl RelayerPort for MaybeRelayerAdapter {
+    async fn await_until_if_in_range(
+        &self,
+        da_height: &DaBlockHeight,
+        max_da_lag: &DaBlockHeight,
+    ) -> anyhow::Result<()> {
+        #[cfg(feature = "relayer")]
+        {
+            if let Some(sync) = self.relayer_synced.as_ref() {
+                let current_height = sync.get_finalized_da_height()?;
+                anyhow::ensure!(
+                    current_height.abs_diff(**da_height) <= **max_da_lag,
```

Comment:
I would say that `da_height - current_height <= max_da_lah` is required, but if `current_height > da_height`, we can instantly return without any error.